morph_tool/axon_point.py

```diff
@@ -0,0 +1,63 @@
+"""Module to detect the terminal point of the main axon/"""
+import logging
+import numpy as np
+
+from morphio import SectionType, IterType
+
+logger = logging.getLogger("morph_tool")
+
+
+def axon_point_section(morph, direction=None, bbox=None):
+    """Estimate axon point as the terminal point of the main axon.
+
+    This point is defined as the point for which the sum of the angles with the direction vector
+    is minimal. This corresponds to a main axon being a branch that follows best a straight line
+    in the given direction. More constraints can be added with bbox argument, if incorect axon is
+    detected.
+
+    Args:
+        morph (morphio.Morphology): morphology
+        direction (ndarray): estimated direction of main axon, if None [0, -1, 0] is used
+        bbox (dict): bbox of the form {'x': [-100, 100], 'y': [-500, -400]}
+
+    Returns:
+        MorphIO section ID for which the last point is the axon point
+    """
+    if direction is None:
+        direction = [0.0, -1.0, 0.0]
+    else:
+        direction /= np.linalg.norm(direction)
+
+    def _get_angle(section, direction):
+        """Get angle between section endpoints and direction."""
+        return np.arccos(
+            np.dot(direction, section.points[-1] - section.points[0])
+            / (np.linalg.norm(section.points[-1] - section.points[0]))
+        )
```

Comment:
Isn't it already coded elsewhere? I feel like this kind of computation is quite usual.

Reply:
couldn't find in numpy/scipy, I didn't look further

Reply:
I meant somewhere else in morph_tool or another BBP package.

Reply:
ah ok, I've never seen that done, but it's no big deal, no?

Reply:
Not at all, I just wondered whether it was duplicated code or not.